pennylane/ops/qubit/special_unitary.py

```diff
@@ -298,32 +434,240 @@ def compute_matrix(theta, num_wires):
+
+    def decomposition(self):
+        r"""Representation of the operator as a product of other operators.
+
+        .. math:: O = O_1 O_2 \dots O_n
 
         Returns:
             list[Operator]: decomposition of the operator
 
         **Example:**
 
         >>> theta = np.array([0.5, 0.1, -0.3])
-        >>> qml.SpecialUnitary.compute_decomposition(theta, 0, num_wires=1)
+        >>> qml.SpecialUnitary(theta, wires=[0]).decomposition()
         [QubitUnitary(array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
             [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]]), wires=[0])]
         """
-        return [qml.QubitUnitary(qml.SpecialUnitary.compute_matrix(theta, num_wires), wires=wires)]
+        theta = self.data[0]
+        if qml.math.requires_grad(theta):
+            interface = qml.math.get_interface(theta)
+            # Get all Pauli words for the basis of the Lie algebra for this gate
+            words = pauli_basis_strings(self.hyperparameters["num_wires"])
+
+            # Compute the linear map that transforms between the Pauli basis and effective generators
+            # Consider the mathematical derivation for the prefactor 2j
+            omega = qml.math.real(2j * self.get_one_parameter_coeffs(interface))
+
+            # Create zero parameters for each Pauli rotation gate that take over the trace of theta
+            detached_theta = _detach(theta, interface)
+            zeros = theta - detached_theta
+            # Apply the linear map omega to the zeros to create the correct preprocessing Jacobian
+            zeros = qml.math.tensordot(omega, zeros, axes=[[1], [0]])
+
+            # Apply Pauli rotations that yield the Pauli basis derivatives
+            paulirots = [
+                TmpPauliRot(zero, word, wires=self.wires, id="SU(N) byproduct")
+                for zero, word in zip(zeros, words)
+            ]
+            return paulirots + [SpecialUnitary(detached_theta, wires=self.wires)]
+
+        return [qml.QubitUnitary(self.matrix(), wires=self.wires)]
 
     def adjoint(self):
         return SpecialUnitary(-self.data[0], wires=self.wires)
+
+
+class TmpPauliRot(PauliRot):
+    r"""A custom version of qml.PauliRot that is inserted with rotation angle zero when
+    decomposing SpecialUnitary. The differentiation logic makes use of the gradient
+    recipe of qml.PauliRot, but deactivates the matrix property so that a decomposition
+    of differentiated tapes is forced. During this decomposition, this private operation
+    removes itself if its angle remained at zero.
+
+    For details see :class:`~.PauliRot`.
+
+    .. warning::
+
+        Do not add this operation to the supported operations of any device.
+        Wrong results and/or severe performance degradations may result.
+    """
+
+    # Deactivate the matrix property of qml.PauliRot in order to force decomposition
+    has_matrix = False
+
+    @staticmethod
+    def compute_decomposition(theta, wires, pauli_word):
```

Comment:
Is it ever used with `theta!=0`?

Reply:
Yep: 
- `SpecialUnitary` creates many `TmpPauliRot` operations at zero
- A gradient transform like `param_shift` moves _one_ of the many operations to $\pm\pi/2$
- All but the moved `TmpPauliRot` remove themselves. The moved `TmpPauliRot` decomposes itself to a `PauliRot`.

Reply:
Great details thanks !